src/clj/rems/db/user_settings.clj

```diff
@@ -74,3 +77,25 @@
 
 (defn delete-user-settings! [user]
   (db/delete-user-settings! {:user user}))
+
+(defn generate-ega-api-key! [user access-token]
+  (assert user "User missing!")
+  (let [config (find-first (comp #{:ega} :type) (:entitlement-push env))] ; XXX: limitation of one type EGA configuration
+    (assert (seq config) "EGA entitlement push must be configured!")
+    (let [response (ega/generate-api-key {:userid user
+                                          :access-token access-token
+                                          :config config})]
+      (when-let [expiration-date (:api-key-expiration-date response)] ; when success
+        (update-user-settings! user {:ega {:api-key-expiration-date expiration-date}}))
```

Comment:
Why does this layer do the `update-user-settings!` but the `update-user-secrets!` is in `rems.ega`? I would've expected to see the calls right next to each other. Perhaps `rems.db.user-settings/generate-ega-api-key!` could be folded into `rems.ega/generate-api-key!` (and ditto for delete)

Reply:
This is a good question, thought about it a bit earlier. I can see it being done in the service layer but there wasn't one for users IIRC? I think I'll refactor it so that there is a clear service for user settings that delegates this stuff to ˋrems.egaˋ. So it will not stay here.

Reply:
I guess it could be ˋrems.egaˋ as the service for EGA things too.

Reply:
Yeah service in `rems.ega`, or a separate `rems.service.ega` both are fine by me

Reply:
So I ended up with a service anyway (that user-settings API delegates to). And a new layer called `ext` which is the same level as `db` but for other external stuff.